code/modules/projectiles/guns/flamer.dm

```diff
@@ -505,6 +500,31 @@
 	if(istype(M))
 		M.flamer_fire_crossed(burnlevel, firelevel)
 
+/mob/living/proc/IsFireResistant()
+	return FALSE
+
+/mob/living/carbon/human/IsFireResistant()
```

Comment:
This helps to clean up the "should i be set on fire code".

Mobs that are resistant still will take dmg though

Reply:
Could you turn this into a trait?

Reply:
I dont think a trait captures the logic

Currently if you have >100 fire resist on a suit you are resistant
If that fails you need to have > 100 on all of (chest, gloves, feet, and helmet)

Reply:
Check how much resistance there is when something with resistance gets added or removed and add or remove the trait when it changes.
You can do this by either storing how much resistance there is at any time or by scanning everything every time. I'd do the former, but even the latter is better than scanning it all every time you want to know if the mob is resistant to fire or not.